src/Migrations/M025ReplaceDeprecatedColorsFromContent.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+namespace P4\MasterTheme\Migrations;
+
+use P4\MasterTheme\MigrationRecord;
+use P4\MasterTheme\MigrationScript;
+
+/**
+ * Replace the deprecated pallete colors from content
+ */
+class M025ReplaceDeprecatedColorsFromContent extends MigrationScript
+{
+    /**
+     * Perform the actual migration.
+     *
+     * @param MigrationRecord $record Information on the execution, can be used to add logs.
+     * phpcs:disable SlevomatCodingStandard.Functions.UnusedParameter -- interface implementation
+     */
+    protected static function execute(MigrationRecord $record): void
+    {
+        global $wpdb;
+
+        $colors = array(
+            [ 'from' => 'grey-80', 'to' => 'grey-900' ],
+            [ 'from' => 'grey-60', 'to' => 'grey-800' ],
+            [ 'from' => 'grey-40', 'to' => 'grey-600' ],
+            [ 'from' => 'grey-20', 'to' => 'grey-500' ],
+            [ 'from' => 'grey-10', 'to' => 'grey-200' ],
+            [ 'from' => 'grey-05', 'to' => 'grey-100' ],
+            [ 'from' => 'gp-green', 'to' => 'green-500' ],
+            [ 'from' => 'dark-blue', 'to' => 'p4-dark-green-800' ],
+            [ 'from' => 'orange-hover', 'to' => 'green-500' ],
+            [ 'from' => 'grey', 'to' => 'grey-900' ],
+            [ 'from' => 'blue', 'to' => 'gp-green-800' ],
+        );
+
+        foreach ($colors as $color) {
+            $from = $color['from'];
+            $to = $color['to'];
+
+            echo sprintf("Replace all the occurrences of '%s' with '%s'\n", $from, $to);
+
+            $sql = '
+                SELECT
+                    ID,
+                    post_parent,
+                    MAX(ID) as latest_post,
+                    count(ID) as total_revisions
+                FROM wp_posts
+                WHERE (
+                    post_content LIKE "%%has-%1$s-background-color%%"
+                    OR post_content LIKE "%%has-%2$s-color%%"
+                )
+                GROUP BY post_parent
+            ';
+
+            $prepared_sql = $wpdb->prepare($sql, [ $from, $from ]);
+            $results = $wpdb->get_results($prepared_sql);
+
+            if (count($results) <= 0) {
+                continue;
+            }
+
+            $posts = json_decode(json_encode($results), true);
```

Comment:
is it to change $results format ?

Reply:
Yes, assuming what you want is an array of arrays instead of an array of objects, you can use the `ARRAY_A` flag for the [`get_results`](https://developer.wordpress.org/reference/classes/wpdb/get_results/#parameters) method, so `$results = $wpdb->get_results($prepared_sql, \ARRAY_A);` on l.58

Reply:
Done!